confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,701 @@
+        return self._rest_client.get('subjects/{}/versions'.format(_urlencode(subject_name)))
+
+    def delete_version(self, subject_name, version):
+        """
+        Deletes a specific version registered to ``subject_name``.
+
+        Args:
+            subject_name (str) Subject name
+            version (int): Version number
+
+        Returns
+            int: Version number which was deleted
+
+        Raises:
+            SchemaRegistryError if the subject or version cannot be found.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#delete--subjects-(string-%20subject)-versions-(versionId-%20version)
+
+        """  # noqa: E501
+        response = self._rest_client.delete('subjects/{}/versions/{}'.
+                                            format(_urlencode(subject_name),
+                                                   version))
+        return response
+
+    def set_compatibility(self, subject_name=None, level=None):
+        """
+        Update global or subject level compatibility level.
+
+        Args:
+            level (str): Compatibility level. See API reference for a list of
+                valid values.
+            subject_name (str, optional): Subject to update. Sets compatibility
+                level policy if not set.
+
+        Returns:
+            str: The newly configured compatibility level.
+
+        Raises:
+            SchemaRegistryError: If the compatibility level is invalid.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#put--config
+
+        """  # noqa: E501
+        if level is None:
+            raise ValueError("level must be set")
+
+        if subject_name is None:
+            return self._rest_client.put('config',
+                                         body={'compatibility': level.upper()})
+
+        return self._rest_client.put('config/{}'
+                                     .format(_urlencode(subject_name)),
+                                     body={'compatibility': level.upper()})
+
+    def get_compatibility(self, subject_name=None):
+        """
+        Get the current compatibility level.
+
+        Args:
+            subject_name (str, optional): Subject name. Returns global policy
+                if left unset.
+
+        Returns:
+            str: Compatibility level for the subject if set, otherwise the
+                global compatibility level.
+
+        Raises:
+            SchemaRegistryError: if the request was unsuccessful.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--config
+
+        """  # noqa: E501
+        if subject_name is not None:
+            url = 'config/{}'.format(_urlencode(subject_name))
+        else:
+            url = 'config'
+
+        return self._rest_client.get(url)
+
+
+class Schema(object):
```

Comment:
this is used as the key in `_SchemaCache.schema_index` so should implement `__hash__`